injectproxy/routes_test.go

```diff
@@ -336,6 +338,33 @@ func TestMatch(t *testing.T) {
 			},
 			expBody: okResponse,
 		},
+		{
```

Comment:
Could you add a test case for the degenerate case where regex matches empty string?